### DISCUSSION TITLE
I can't install ZIM

### DISCUSSION THREAD

**Discussion by PILIHU2022** [zimfw/zimfw#506] [Q&A] [answered]
Hello developer, I am from China, I want to install power10k in Z SH through ZI M, but it prompts this, how can I solve this problem?

) Using Zsh version 5.9
) ZIM_HOME not set, using the default one.
) Zsh is your default shell.
) Downloaded the Zim script to /home/***/.zim/zimfw.zsh
) Prepended Zim template to /home/***/.zimrc
) Prepended Zim template to /home/***/.zshrc
x utility: Error during git clone
  致命错误：无法访问 'https://github.com/zimfw/utility.git/'：OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443 
x completion: Error during git clone
  致命错误：无法访问 'https://github.com/zimfw/completion.git/'：OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443 
x /home/PILIHU/.zimrc:20:utility: /home/***/.zim/modules/utility not found
x /home/PILIHU/.zimrc:41:completion: /home/***/.zim/modules/completion not found
x /home/PILIHU/.zimrc:76:utility: /home/***/.zim/modules/utility not found
x /home/PILIHU/.zimrc:97:completion: /home/***/.zim/modules/completion not found
x /home/PILIHU/.zimrc:132:utility: /home/***/.zim/modules/utility not found
x /home/PILIHU/.zimrc:153:completion: /home/***/.zim/modules/completion not found
Failed to source /home/***/.zimrc
x Could not install modules.

**Comment by antoineco** [comment]
It seems like you are not allowed to clone things from GitHub.

What is the full output of `git clone https://github.com/zimfw/zimfw.git`?

**Reply by ericbn** [reply, marked as answer]
Looks like China blocks access to GitHub. https://en.wikipedia.org/wiki/Censorship_of_GitHub#China

**Comment by ericbn** [comment]
How did you download the install script?

Did `curl -fsSL https://raw.githubusercontent.com/zimfw/install/master/install.zsh | zsh` works for you? If yes, then follow these steps:
1. Add `zstyle ':zim:zmodule' use 'degit'` to your .zshrc file
2. Restart your terminal
3. Remove all installed modules with `rm -rf ${ZIM_HOME}/modules`
4. Reinstall the modules with `zimfw install`

Not sure if this will work, but might be worth trying.

EDIT: This solution will try to install modules using api.github.com instead of github.com. Hopefully the former is not blocked as the latter.

**Comment by skt256** [comment]
挂个科学上网
